resources/network/virtual_network.go

```diff
@@ -0,0 +1,320 @@
+			SubscriptionID: subscriptionID,
+			IP:             value,
+		}
+		tValues = append(tValues, &tValue)
+	}
+	return tValues
+}
+
+func transformVirtualNetworkIPAllocations(subscriptionID string, values *[]network.SubResource) []*VirtualNetworkIPAllocation {
+	var tValues []*VirtualNetworkIPAllocation
+	for _, value := range *values {
+		tValue := VirtualNetworkIPAllocation{
+			SubscriptionID: subscriptionID,
+			ResourceID:     value.ID,
+		}
+		tValues = append(tValues, &tValue)
+	}
+	return tValues
+}
+
+func transformVirtualNetworkPeerings(subscriptionID string, values *[]network.VirtualNetworkPeering) []*VirtualNetworkPeering {
+	var tValues []*VirtualNetworkPeering
+	for _, value := range *values {
+		tValue := VirtualNetworkPeering{
+			SubscriptionID:         subscriptionID,
+			RemoteVirtualNetworkID: value.VirtualNetworkPeeringPropertiesFormat.RemoteVirtualNetwork.ID,
+		}
+		tValues = append(tValues, &tValue)
+	}
+	return tValues
+}
+
+func transformVirtualNetworkSubnets(subscriptionID string, values *[]network.Subnet) []*Subnet {
+	var tValues []*Subnet
+	for _, value := range *values {
+		tValue := Subnet{
+			SubscriptionID:                    subscriptionID,
+			ResourceID:                        value.ID,
+			Name:                              value.Name,
+			Etag:                              value.Etag,
+			Purpose:                           value.Purpose,
+			AddressPrefix:                     value.AddressPrefix,
+			AddressPrefixes:                   transformSubnetAddressPrefixes(subscriptionID, value.AddressPrefixes),
+			ProvisioningState:                 string(value.ProvisioningState),
+			PrivateEndpointNetworkPolicies:    value.PrivateEndpointNetworkPolicies,
+			PrivateLinkServiceNetworkPolicies: value.PrivateLinkServiceNetworkPolicies,
+			NatGatewayID:                      value.NatGateway.ID,
+		}
+		tValues = append(tValues, &tValue)
+	}
+	return tValues
+}
+
+func transformSubnetAddressPrefixes(subscriptionID string, values *[]string) []*SubnetAddressPrefix {
+	var tValues []*SubnetAddressPrefix
+	for _, value := range *values {
+		tValue := SubnetAddressPrefix{
+			SubscriptionID: subscriptionID,
+			AddressPrefix:  value,
+		}
+		tValues = append(tValues, &tValue)
+	}
+	return tValues
+}
+
+type VirtualNetworkConfig struct {
+	Filter string
+}
+
+var VirtualNetworkTables = []interface{}{
+	&VirtualNetwork{},
+	&VirtualNetworkTag{},
+}
+
+func VirtualNetworks(subscriptionID string, auth autorest.Authorizer, db *database.Database, log hclog.Logger, gConfig interface{}) error {
+	var config VirtualNetworkConfig
+	ctx := context.Background()
+	err := mapstructure.Decode(gConfig, &config)
+	if err != nil {
+		return err
+	}
+
+	svc := network.NewVirtualNetworksClient(subscriptionID)
+	svc.Authorizer = auth
+	output, err := svc.ListAll(ctx)
+	if err != nil {
+		return err
+	}
+
+	values := make([]network.VirtualNetwork, 0)
+	for output.NotDone() {
+		values = append(values, output.Values()...)
```

Comment:
It is better to transform inside the loop and put every chunk straight into the database. Otherwise it can eat up all the memory pretty quickly.